frontend/app/utils/ReferralData.scala

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import play.api.mvc.{Cookie, RequestHeader}
+
+case class ReferralData(campaignCode: Option[String], url: Option[String], pageviewId: Option[String])
+
+object ReferralData {
+
+  val CampaignCodeKey = "mem_campaign_code"
+  val UrlKey = "gu_mem_ref_url"
+  val PageviewIdKey = "gu_refpvid"
+
+  def makeCookies(implicit request: RequestHeader): Seq[Cookie] = {
+    val refUrl = request.headers.get("referer").map(Cookie(ReferralData.UrlKey, _))
```

Comment:
this is causing the 500 error pagerduty alerts as not everything that can be in a referer header is valid in a cookie value